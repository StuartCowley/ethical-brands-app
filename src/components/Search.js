import React from "react";
import { Link } from "react-router-dom";
import "../styles/search.css";
import Alert from "./Alert";

const Search = () => {
  return (
    <div className="search-component">
      <Alert message="Not in database" success={false} />
      <h2 className="search-title">Ethical Brands</h2>
      <form className="search-form">
        <div>
          <input type="text" className="search-input" />
        </div>
        <div>
          <button type="submit" className="search-button">
            <Link to="/search-results">SEARCH</Link>
          </button>
        </div>
      </form>
    </div>
  );
};

export default Search;
