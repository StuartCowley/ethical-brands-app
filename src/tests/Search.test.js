import { render, screen } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import Search from "../components/Search";

describe("search", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Search />
      </BrowserRouter>
    );
    const search = asFragment();

    expect(search).toMatchSnapshot();
  });

  it("checks search button is rendered correctly", () => {
    render(
      <BrowserRouter>
        <Search />
      </BrowserRouter>
    );
    const linkElement = screen.getByText(/search/i);

    expect(linkElement).toBeInTheDocument();
  });
});
