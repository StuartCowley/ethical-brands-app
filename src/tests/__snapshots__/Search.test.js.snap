// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search snapshot test 1`] = `
<DocumentFragment>
  <div
    class="search-component"
  >
    <h2
      class="search-title"
    >
      Ethical Brands
    </h2>
    <form
      class="search-form"
    >
      <div>
        <input
          class="search-input"
          type="text"
        />
      </div>
      <div>
        <button
          class="search-button"
          type="submit"
        >
          <a
            href="/search-results"
          >
            SEARCH
          </a>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
